File: easy/road_trip.rb
File.open(ARGV[0]).each_line do |line|
  s = line.gsub(/[^\d ]/, '').split.map(&:to_i).sort
  (-s.length + 1..-1).each { |i| s[-i] -= s[-i - 1] }
  puts s.join(',')
end

File: easy/road_trip.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    $line =~ s/[A-Z][a-z]+,|;//g;
    my $s = 0;
    my @r = map {
        my $t = $_ - $s;
        $s += $t;
        $t;
    } sort { $a <=> $b } split( / /, $line );
    printf( "%s\n", join( ',', @r ) );
}
close(INFILE);
